tests/scripts/translate_ciphers.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+
+# translate_ciphers.py
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Translate ciphersuite names in MBedTLS format to OpenSSL and GNUTLS
+standards.
+
+Format and analyse strings past in via input arguments to match
+the expected strings utilised in compat.sh.
+
+sys.argv[1] should be "g" or "o" for GNUTLS or OpenSSL.
+sys.argv[2] should be a string containing one or more ciphersuite names.
+"""
+
+import re
+import sys
+
+def translate_gnutls(m_cipher):
```

Comment:
I thought about this but I thought substituting substrings would be more reliable and easier to read, especially when it came to gnutls as that uses `:+` twice for each ciphersuite. I could help in a lot of places where you want to check the proceeding block, but then it could also be less readable when you do not know the index of something e.g `"WITH"`, where you would have to get the index first, then deal with it